gemhardware/include/gem/hw/GEMHwDevice.h

```diff
@@ -6,29 +6,32 @@
 #include <iomanip>
 #include <memory>
 
-//#include "xdata/InfoSpace.h"
-/* #include "xdata/InfoSpaceFactory.h" */
-
-#include "xdata/String.h"
-#include "xdata/UnsignedLong.h"
-#include "xdata/UnsignedInteger32.h"
+/* #include "xdata/String.h" */
+/* #include "xdata/UnsignedLong.h" */
+/* #include "xdata/UnsignedInteger32.h" */
+// can we decouple completely from these?
 #include "toolbox/string.h"
-#include "toolbox/mem/Reference.h"
-#include "toolbox/mem/Pool.h"
+/* #include "toolbox/mem/Reference.h" */
+/* #include "toolbox/mem/Pool.h" */
 
 #include "uhal/uhal.hpp"
 /* #include "uhal/Utilities.hpp"  /\* removed in 2.5 or 2.6 *\/ */
 #include "uhal/utilities/bits.hpp"
 #include "uhal/utilities/files.hpp"
 
+#include "xhal/XHALInterface.h"
+/* #include "xhal/XHALDevice.h" */
+
+#include "gem/hw/utils/GEMCrateUtils.h"
+#include "gem/hw/exception/Exception.h"
+
+/// TODO: would be good to decouple from utils, maybe migrate RegisterUtils to gem::hw::utils
 #include "gem/utils/GEMLogging.h"
 #include "gem/utils/GEMRegisterUtils.h"
-
+// TODO: This is a generically useful lock, both inside and outside of the HW, but is it used in the HW yet?
```

Comment:
Why is it better than `std::mutex`/`std::lock_guard`?

Reply:
It was brought from some other `xdaq` based code and wraps an `xdaq` semaphore.
There it *may* have been implemented in this way due to compiler constraints, so would have to go through our other uses of this `BSem` object and see if it would make sense to replace them all with `std::` utilities